### PR TITLE
KQL DateTime functions PART 1

### PR DESCRIPTION
**Pull request by HeenaBansal2009**
### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
This PR implements dateTime functions for KQL.
https://zenhub.ibm.com/workspaces/clickhouse-project-61250df53aaf060db4e08052/issues/clickhouse/issue-repo/724

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/Parsers/Kusto/KustoFunctions/KQLDateTimeFunctions.cpp [hunk 1]
```diff
@@ -63,23 +63,23 @@ bool DatetimeDiff::convertImpl(String &out,IParser::Pos &pos)
...
 bool DayOfWeek::convertImpl(String &out,IParser::Pos &pos)
 {
-    String res = String(pos->begin,pos->end);
-    out = res;
-    return false;
+    const String fn_name = getKQLFunctionName(pos);
+    if (fn_name.empty())
+        return false;
+    ++pos;
+
```
Comment (kashwy) [inline src/Parsers/Kusto/KustoFunctions/KQLDateTimeFunctions.cpp:75]: missing an argument ?

Comment (HeenaBansal2009) [reply src/Parsers/Kusto/KustoFunctions/KQLDateTimeFunctions.cpp:75]: > missing an argument ?

Ohh Yes, Copy paste error to new branch.I'll update it.

File: src/Parsers/Kusto/KustoFunctions/KQLDateTimeFunctions.cpp [hunk 2]
```diff
@@ -162,39 +156,91 @@ bool Now::convertImpl(String &out,IParser::Pos &pos)
...
+    out = std::format("date_add(Week,{}, toDateTime64((toStartOfWeek({})) , 9 , 'UTC')) ", offset, datetime_str);
+    return true;
 }
 
 bool StartOfYear::convertImpl(String &out,IParser::Pos &pos)
 {
-    String res = String(pos->begin,pos->end);
-    out = res;
-    return false;
+       const String fn_name = getKQLFunctionName(pos);
```
Comment (kashwy) [inline src/Parsers/Kusto/KustoFunctions/KQLDateTimeFunctions.cpp:229]: align the start of line

File: src/Parsers/Kusto/KustoFunctions/KQLDateTimeFunctions.cpp [hunk 3]
```diff
@@ -220,16 +266,32 @@ bool UnixTimeNanosecondsToDateTime::convertImpl(String &out,IParser::Pos &pos)
 
 bool UnixTimeSecondsToDateTime::convertImpl(String &out,IParser::Pos &pos)
 {
-    String res = String(pos->begin,pos->end);
-    out = res;
-    return false;
+       const String fn_name = getKQLFunctionName(pos);
```
Comment (kashwy) [inline src/Parsers/Kusto/KustoFunctions/KQLDateTimeFunctions.cpp:269]: align the start of line

**Review by kashwy — APPROVED** [review]

**Pull request merged**